### PR TITLE
Update the build ID and job URL in prow job spec when using the exter…

### PR DESCRIPTION
**Pull request by ccojocar**
…nal pipelinerunner

Also use patch updating the prow job status.

### CODE REVIEW

File: prow/cmd/pipeline/controller.go [hunk 1]
```diff
@@ -806,3 +831,13 @@ func (c *controller) requestPipelineRun(context, namespace string, pj prowjobv1.
...
 }
+
+func getBuildID(pr *pipelinev1alpha1.PipelineRun) string {
+	for _, param := range pr.Spec.Params {
+		if param.Name == "build_id" {
+			return param.Value
+		}
+	}
+	// use a default build number if the real build number cannot be parsed
+	return "1"
```
Comment (pmuir) [inline prow/cmd/pipeline/controller.go:848]: Should this be `0` or `9999999` to avoid it looking like a real number?

Comment (ccojocar) [reply prow/cmd/pipeline/controller.go:848]: @pmuir It seems that `jx` defaults to `1`. https://github.com/jenkins-x/jx/blob/24cfb8086af95a2187699ecebd5ba2c42527b651/pkg/jx/cmd/step_create_task.go#L304

File: prow/cmd/pipeline/controller.go [hunk 2]
```diff
@@ -806,3 +831,13 @@ func (c *controller) requestPipelineRun(context, namespace string, pj prowjobv1.
 
 	return "", fmt.Errorf("no PipelineRun object found returned by pipeline runner")
 }
+
+func getBuildID(pr *pipelinev1alpha1.PipelineRun) string {
+	for _, param := range pr.Spec.Params {
+		if param.Name == "build_id" {
```
Comment (dwnusbaum) [inline prow/cmd/pipeline/controller.go:843]: FWIW based on my read of https://github.com/jenkins-x/jx/blob/7a77c9f2b0d6c233fb8c0e49323ca7a270da0d8e/pkg/jx/cmd/step_create_task.go#L1414 and the rest of the code in that file, I _think_ this param will only be set on release pipelines currently, but I'm not entirely sure.

Comment (ccojocar) [reply prow/cmd/pipeline/controller.go:843]: I have seen that is also set on a pull request PipelineRun. Thanks for bringing this up. I'll check it.

**Pull request merged**